arch/arm/src/stm32f7/Kconfig

```diff
@@ -2136,6 +2136,14 @@ endchoice #"ULPI Selection"
 
 endmenu # OTG_HS Config
 
+config STM32F7_EXTERNAL_RAM
+    bool "External RAM on FMC"
+    default n
+    depends on STM32F7_FMC
+    select ARCH_HAVE_HEAP2
+    ---help---
+        In addition to internal SDRAM, external RAM may be available through the FMC.
+
```

Comment:
Where is this new configuration option used?  It is not used in arch/arm/src/stm32f7 so it does not belong there.

Do you use it in your board?  If so then it belongs in your boards/arm/stm32/<board>/Kconfig

Reply:
The sole purpose of STM32F7_EXTERNAL_RAM is to select ARCH_HAVE_HEAP2.
ARCH_HAVE_HEAP2 is used in arch/arm/src/stm32f7/stm32_allocateheap.c
There is no other way to select ARCH_HAVE_HEAP2.

It is the same as in stm32h7. Stm32H7 has STM32H7_EXTERNAL_RAM, which just selects ARCH_HAVE_HEAP2.
And Stm32 also has this option. STM32_EXTERNAL_RAM
Only Stm32F7 is missing it.

Reply:
In our own board, which also uses F7, we also had to hack around `ARCH_HAVE_HEAP2` to get external RAM to work.
So I think there's some value to this change, if this were merged, we would switch to this option.